src/main/java/lab2/AvatarGenerator.java

```diff
@@ -1,66 +1,76 @@
-        }
 
+            var avatarStream = AvatarGenerator.getRandomAvatarStream(); // Call to getRandomAvatarStream to fetch a random avatar
+            // avatarString returns an InputStream, object reference type
+            AvatarGenerator.showAvatar(avatarStream); // Call to showAvatar with the fetched InputStream to display the avatar
+        } catch (IOException | InterruptedException e) { // Catch IOException or InterruptedException
+            e.printStackTrace(); // Print the stack trace for debugging
+        }
     }
 
     public static InputStream getRandomAvatarStream() throws IOException, InterruptedException {
-        // Pick a random style
-        String[] styles = { "adventurer", "adventurer-neutral", "avataaars", "big-ears", "big-ears-neutral", "big-smile", "bottts", "croodles", "croodles-neutral", "fun-emoji", "icons", "identicon", "initials", "lorelei", "micah", "miniavs", "open-peeps", "personas", "pixel-art", "pixel-art-neutral" };
-        var style = styles[(int)(Math.random() * styles.length)];
+        // generates a random avatar stream. Throws IOException and InterruptedException
+
+        String[] styles = { "adventurer", "adventurer-neutral", "avataaars", "big-ears", "big-ears-neutral", "big-smile"
+                , "bottts", "croodles", "croodles-neutral", "fun-emoji", "icons", "identicon", "initials", "lorelei",
+                "micah", "miniavs", "open-peeps", "personas", "pixel-art", "pixel-art-neutral" }; // styles is an array of strings
+        // Randomly selects one one avatar style from the array "styles"
+        var style = styles[(int)(Math.random() * styles.length)]; // "style" is a String
 
-        // Generate a random seed
-        var seed = (int)(Math.random() * 10000);
+        // Generates the random seed for the avatar HTTP request
+        var seed = (int)(Math.random() * 10000); // 'seed' is a primitive type, integer value
 
-        // Create an HTTP request for a random avatar
+        // Create an HTTP request URI to fetch a random avatar with a seed
         var uri = URI.create("https://api.dicebear.com/9.x/%s/png?seed=%d".formatted(style, seed));
+        // uri returns a java.net.URI type, reference type
+
+        // Create an HttpRequest object with the generated URI
         var request = HttpRequest.newBuilder(uri).build();
+        // request returns a java.net.http.HttpRequest type, reference type
 
-        // Send the request
-        try (var client = HttpClient.newHttpClient()) {
-            var response = client.send(request, HttpResponse.BodyHandlers.ofInputStream());
-            return response.body();
+        // send the HTTP request and get the avatar image as InputStream
+        try (var client = HttpClient.newHttpClient()) { // create a new HttpClient instance to send the HTTP request
+            var response = client.send(request, HttpResponse.BodyHandlers.ofInputStream());// sends the request and receives the response as an InputStream
```

Comment:
Again, at each `.` you should be identifying whether the identifier is a class/instance method/variable